cmd/execution/main.go

```diff
@@ -178,7 +180,7 @@ func main() {
 			if err != nil {
 				return nil, fmt.Errorf("cannot create checkpointer: %w", err)
 			}
-			compactor := wal.NewCompactor(checkpointer, 10*time.Second, checkpointDistance)
+			compactor := wal.NewCompactor(checkpointer, 10*time.Second, checkpointDistance, checkpointsToKeep)
```

Comment:
this is great, do we have a fallback mechanism to reject a checkpoint and fetch the older one?

Reply:
I'm not sure I get what you mean? Cleaning up checkpoints simply removes old ones